### ISSUE TITLE
Moderator on moderated.jitsi.net not supported (redirect issue)

### ISSUE THREAD

**Issue by OrionRandD** [jitsi/jitsi-meet-electron#705]
How can I have moderator powers in jitsi-meet-electron?
Do you provide this feature?

https://moderated.jitsi.net/

If not, could you add it?
Thx

**Comment by saghul** [comment]
Good point. We probably need to change things around and do a HEAD request to resolve the real URL... And then parse the jwt and pass it along.

A PR in this space would be welcome.

**Comment by OrionRandD** [comment]
Thx for replying. Could you make that PR for me?
I do now know how to do that...
Since I, and yourself, seem to agree in this point here.

Another suggestion I would give is NOT to allow jitsi-electron to send
an invitation to join a certain room without the person who opened the room to set a password before sending the invitation...

-------------------------------

**Comment by saghul** [comment]
> Thx for replying. Could you make that PR for me? I do now know how to do that... Since I, and yourself, seem to agree in this point here.

Will do when I have some time.



> Another suggestion I would give is NOT to allow jitsi-electron to send an invitation to join a certain room without the person who opened the room to set a password before sending the invitation...

There is no reason to enforce that, it's the user's responsibility.

**Comment by csett86** [comment]
While revisiting this and checking how moderated.jitsi.net works, I don't see a way how to enable this into the electron app.

moderated.jitsi.net hides the real URL with JWT and only redirects to it when clicking a button: 

![Bildschirmfoto 2022-07-24 um 15 07 39](https://user-images.githubusercontent.com/1392875/180648417-e8350bc9-64dd-488f-a131-130fa887156b.png)

Source: https://github.com/jitsi/moderated-meetings/blob/master/src/main/js/screens/Join.tsx#L152-L157

So unless this is changed in moderated.jitsi.net to directly expose the URL containing the JWT, I don't see how this would work in the electron app. Any suggestions @saghul?

**Comment by saghul** [comment]
@csett86 Yep, you're right. We'd need to add a way to copy the real link.

**Comment by csett86** [comment]
@saghul I could provide an updated implementation as PR for the moderated-meetings that looks like this:

![Bildschirmfoto 2022-07-29 um 18 28 07](https://user-images.githubusercontent.com/1392875/181803698-bd9c92e7-4618-443e-b601-476b26bab510.png)

This changes the room names to url sanitized base64 (ie "/" -> "-" and "+" -> "_") names so that the room names are shorter and then extends the view to 720px so that the normal and moderator URL also look different.

What do you think?

**Comment by csett86** [comment]
Its not a good idea, it would break all existing moderated meetings by assigning them a new room. I‘ll try a different route

**Comment by saghul** [comment]
Thanks for looking into this.

**Comment by csett86** [comment]
I think I got something now that allows to share the moderation link and also keeps backwards compatibility:

https://github.com/jitsi/moderated-meetings/pull/24
